elastic/kibana
# [Response Ops] Fixing scripted metric agg in alerting telemetry

Resolves https://github.com/elastic/kibana/issues/127837

## Summary

* Fixes the reduce section of all the scripted metric aggs in the alerting telemetry. Each reduce section assumed it was receiving something like a `Record<string, number>` from each shard, so it would iterate through each key of the record and try to sum the values. However, it seems that all of our scripted metrics are grouping by the rule type ID, so the script was actually returning something like `Record<string, Record<string, number>>`, which would fail during the `reduce` since it was trying to add two records together. This issue rarely shows up during local development since all our `.kibana` and event log indices are on one shard.
* Adds a try/catch to each aggregation run while collecting alerting telemetry. This way, if one aggregation fails, it does not prevent the other aggregations from return data. Also tries to log the full error message.

## To Verify
* To recreate the error
  * Modify the alerting telemetry to run more frequently. Run an older version of ES/Kibana while saving the ES data using `yarn es snapshot --ssl --license trial -E path.data=../path_to_data`. Create some rules and let them run for a bit to populate the event log.
  * Modify the alerting telemetry to run more frequently. Run `main` using the same data as the previous step. You should see an alerting telemetry error logged each time the telemetry task runs. Navigating to `https://localhost:5601/api/stats?extended=true&legacy=true`, you should not see any data for the alerting telemetry.
  * Run this PR branch. You should see the telemetry task run successfully and you should be able to see values in the alerting telemetry.
